src/main/java/com/aws/greengrass/shadowmanager/ipc/InboundRateLimiter.java

```diff
@@ -71,22 +103,18 @@ public void clear() {
     }
 
     /**
-     * Sets the overall inbound rate limiter rate.
-     *
-     * @param rate Max inbound requests per second for all things
+     * get total inbound rate limiter.
+     * @return totalInboundRateLimiter
      */
-    public void setTotalRate(int rate) {
-        totalRate.set(rate);
-        totalInboundRateLimiter.setRate(totalRate.get());
+    public IpcRateLimiter getTotalInboundRateLimiter() {
+        synchronized (totalInboundRateLimiter) {
+            totalInboundRateLimiter.setRate(getTotalRate());
+        }
+        return totalInboundRateLimiter;
```

Comment:
why does this need to be synchronized? `setRate` is already a synchronized method in the rate limiter

also this behavior could be very confusing - especially with no documentation. Calling a getter is going to potentially mutate the rate limiter. 

Is there a performance impact of looking at the config on every request? Why not subscribe to the config topic and only update on changes like in other config based objects?